packages/@aws-cdk/aws-batch/test/compute-environment.test.ts

```diff
@@ -0,0 +1,400 @@
+import { expect, haveResource, haveResourceLike, ResourcePart } from '@aws-cdk/assert';
+import '@aws-cdk/assert/jest';
+import { AmazonLinuxGeneration, InstanceClass, InstanceSize, InstanceType, SecurityGroup, SubnetType, Vpc } from '@aws-cdk/aws-ec2';
+import { AmiHardwareType, EcsOptimizedAmi } from '@aws-cdk/aws-ecs';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { throws } from 'assert';
+import * as batch from '../lib';
+
+const computeResourceProps = (override: batch.ComputeResourceProps): batch.ComputeResourceProps => {
+    return {
+        ...override,
+    };
+};
+
+const expectedManagedDefaultProps = {
+    ServiceRole: {
+        Ref: 'testcomputeenvResourceServiceLinkedRoleDC93CC0B',
+    },
+    Type: batch.ComputeEnvironmentType.MANAGED,
+};
+
+const expectedUnmanagedDefaultComputeProps = (override: any) => {
+    return {
+        ComputeResources: {
+            AllocationStrategy: batch.AllocationStrategy.BEST_FIT,
+            InstanceRole: {
+                'Fn::GetAtt': [
+                    'testcomputeenvResourceRoleBD565AC0',
+                    'Arn'
+                ]
+            },
+            InstanceTypes: [
+                'optimal'
+            ],
+            MaxvCpus: 256,
+            MinvCpus: 0,
+            Subnets: [
+                {
+                    Ref: 'testvpcPrivateSubnet1Subnet865FB50A'
+                },
+                {
+                    Ref: 'testvpcPrivateSubnet2Subnet23D3396F'
+                }
+            ],
+            Type: batch.ComputeResourceType.EC2,
+            ...override,
+        },
+    };
+};
+
+describe('When creating a batch compute evironment', () => {
+    describe('with no properties provided', () => {
+        test('should create an AWS managed environment', () => {
+            // GIVEN
+            const stack = new cdk.Stack();
+
+            // WHEN
+            new batch.ComputeEnvironment(stack, 'test-compute-env');
+
+            // THEN
+            expect(stack).to(haveResourceLike('AWS::Batch::ComputeEnvironment', {
+                Type: batch.ComputeEnvironmentType.MANAGED
+            }, ResourcePart.Properties));
```

Comment:
This is the default, so feel free to skip it.

Reply:
Got it 👍